dowhy/causal_refuters/graph_refuter.py

```diff
@@ -1,69 +1,133 @@
+            if self._data[node].dtype == np.int64 or self._data[node].dtype == np.int32:
+                discrete_columns.append(node)
+                if self._data[node].isin([0,1]).all():
+                    binary_columns.append(node)
+            else:
+                continuous_columns.append(node)
+        for a, b, c in independence_constraints:
+            if a in continuous_columns and b in continuous_columns and all(node in continuous_columns for node in c):
+                # a, b and c are all continuous variables
+                if self._method_name_continuous is None or self._method_name_continuous == "partial_correlation":
+                    self.partial_correlation(x = a, y= b, z = c)
                 else:
-                    self._true_implications.append([a, b, c])
-                    self._results[key]= [p_value, True]
-            self.set_refutation_result() #Set refutation result accordingly
-            refute.add_conditional_independence_test_result(number_of_constraints_model = len(independence_constraints), number_of_constraints_satisfied = len(self._true_implications), refutation_result = self._refutation_passed)
-        else:
-            self.logger.error("Invalid conditional independence test")
+                    self.logger.error("Invalid conditional independence test for continuous data. Supported tests - partial_correlation")
+
+            elif a in discrete_columns and b in discrete_columns and all(node in discrete_columns for node in c):
+                # a, b and c are all discrete variables
+                if self._method_name_discrete is None or self._method_name_discrete == "conditional_mutual_information":
+                    self.conditional_mutual_information(x = a, y = b, z =c)
+                else:
+                    self.logger.error("Invalid conditional independence test for discrete data. Supported tests - conditional_mutual_information")
+
+            elif all(node in discrete_columns for node in c): 
+                # c is discrete and either a or b is continuous and the other is discrete
+                self.conditional_mutual_information(x = a, y= b, z = c)
+            elif all(node in continuous_columns for node in c) and ((a in continuous_columns and b in binary_columns) or (a in binary_columns and b in continuous_columns) or (a in binary_columns and b in binary_columns) ): 
+                # c is continuous and 
+                #   1. either a and b is continuous and the other is binary
+                #   2. both and b are binary
+                self.partial_correlation(x = a, y= b, z = c)
+            elif all(node in continuous_columns for node in c) and ((a in continuous_columns and b not in binary_columns) or (b in continuous_columns and a not in binary_columns)):
+                # c is continuous and either of a and b is continuous and the other is discrete-non binary
+                print("The following setting not supported")
```

Comment:
instead of "following", can you output the specific setting. E.g., the setting with a as discrete, b as non-discrete, etc.